doc/user/client.md

```diff
@@ -0,0 +1,411 @@
+
+    // PropagationPolicy determined whether and how garbage collection will be
+    // performed. Either this field or OrphanDependents may be set, but not both.
+    // The default policy is decided by the existing finalizer set in the
+    // metadata.finalizers and the resource-specific default policy.
+    // Acceptable values are: 'Orphan' - orphan the dependents; 'Background' -
+    // allow the garbage collector to delete the dependents in the background;
+    // 'Foreground' - a cascading policy that deletes all dependents in the
+    // foreground.
+    PropagationPolicy *metav1.DeletionPropagation
+
+    // Raw represents raw DeleteOptions, as passed to the API server.
+    Raw *metav1.DeleteOptions
+}
+```
+
+```Go
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	pod := &v1.Pod{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, pod)
+
+	...
+
+	ctx := context.TODO()
+	if pod.Status.Phase == v1.PodUnknown {
+		df := func(opts *client.DeleteOptions) {
+			s := 5
+			opts.GracePeriodSeconds = &s // Delete the pod after 5 seconds.
+		}
+		err := r.client.Delete(ctx, pod, df)
+
+		...
+	}
+
+	...
+}
+```
+
+#### Get
+
+```Go
+// Get retrieves an API object for a given object key from the k8s cluster
+// and stores it in obj.
+func (c Client) Get(ctx context.Context, key ObjectKey, obj runtime.Object) error
+```
+
+```Go
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	app := &v1alpha1.App{}
+	ctx := context.TODO()
+	name := request.NamespacedName
+	err := r.client.Get(ctx, name, app)
+
+	...
+}
+```
+
+#### List
+
+```Go
+// List retrieves a list of objects for a given namespace and list options
+// and stores the list in obj.
+func (c Client) List(ctx context.Context, opts *ListOptions, obj runtime.Object) error
+```
+
+```Go
+type ListOptions struct {
+    // LabelSelector filters results by label.  Use SetLabelSelector to
+    // set from raw string form.
+    LabelSelector labels.Selector
+
+    // FieldSelector filters results by a particular field.  In order
+    // to use this with cache-based implementations, restrict usage to
+    // a single field-value pair that's been added to the indexers.
+    FieldSelector fields.Selector
+
+    // Namespace represents the namespace to list for, or empty for
+    // non-namespaced objects, or to list across all namespaces.
+    Namespace string
+
+    // Raw represents raw ListOptions, as passed to the API server.  Note
+    // that these may not be respected by all implementations of interface,
+    // and the LabelSelector and FieldSelector fields are ignored.
+    Raw *metav1.ListOptions
+}
+```
+
+```Go
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+	
+	// Return all pods with a label of request.NamespacedName
+	opts := &client.ListOptions{}
+	opts.SetLabelSelector(request.NamespacedName)
```

Comment:
[request.NamespacedName](https://github.com/kubernetes-sigs/controller-runtime/blob/master/vendor/k8s.io/apimachinery/pkg/types/namespacedname.go#L31-L34) is not the correct input to this helper.
This needs a raw string in the format of a label selector e.g `foo=bar` or `environment in (production, qa)` etc.
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors

We can demonstrate this better via:
```Go
// // Return all pods in the request namespace with a label of `app=<name>`
opts := &client.ListOptions{}
opts.SetLabelSelector(fmt.Sprintf("app=%s", request.NamespacedName.Name))
opts.InNamespace(request.NamespacedName.Namespace)
```